### PR TITLE
docs: add Query documentation

### PR DESCRIPTION
**Pull request by shahednasser**
- Replace remote query documentation with new Query documentation
- Add redirect from old remote query to new query documentation
- Update remote query usages across docs to use new query usage.

### CODE REVIEW

File: www/apps/book/app/advanced-development/modules/query/page.mdx [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+    "id", 
+    "name",
+    "products.*",
+  ],
+})
+```
+
+---
+
+## Apply Filters
```
Comment (olivermrbl) [inline www/apps/book/app/advanced-development/modules/query/page.mdx:124]: question: Should we make it clear that the `graph` engine does not support cross-module filtering?

Comment (olivermrbl) [reply www/apps/book/app/advanced-development/modules/query/page.mdx:124]: Same goes for sorting.

File: www/apps/resources/app/recipes/digital-products/page.mdx [hunk 2]
```diff
@@ -135,9 +135,9 @@ Use the remote link to create a link between two records, and use the remote que
     icon: AcademicCapSolid,
   },
   {
-    href: "!docs!/advanced-development/modules/remote-query",
-    title: "How to Use the Remote Query",
-    text: "Learn how to fetch data across modules with remote query.",
+    href: "!docs!/advanced-development/modules/query",
+    title: "How to Use the Query",
+    text: "Learn how to fetch data across modules with the Query.",
```
Comment (olivermrbl) [inline www/apps/resources/app/recipes/digital-products/page.mdx]: suggestion: Drop the "the". Think it's cleaner

Comment (olivermrbl) [reply www/apps/resources/app/recipes/digital-products/page.mdx]: Or Medusa Query if it doesn't fit well

File: www/apps/book/app/advanced-development/modules/query/page.mdx (hunk 1 above)
Comment (shahednasser) [reply www/apps/book/app/advanced-development/modules/query/page.mdx:124]: Makes sense, will add 👍🏻

File: www/apps/resources/app/recipes/digital-products/page.mdx [hunk 3]
```diff
@@ -123,9 +123,9 @@ Create workflows to implement these flows, then utilize these workflows in other
 
 ## Manage Linked Records
 
-If you've defined links between data models of two modules, you can manage them through two functions: remote link and remote query.
+If you've defined links between data models of two modules, you can manage them through two functions: remote link and Query.
 
-Use the remote link to create a link between two records, and use the remote query to fetch data across linked data models.
+Use the remote link to create a link between two records, and use the Query to fetch data across linked data models.
```
Comment (olivermrbl) [inline www/apps/resources/app/recipes/digital-products/page.mdx]:
```suggestion
Use the remote link to create a link between two records, and use Query to fetch data across linked data models.
```

File: www/apps/resources/app/recipes/marketplace/examples/restaurant-delivery/page.mdx [hunk 4]
```diff
@@ -662,13 +662,11 @@ export async function GET(req: MedusaRequest, res: MedusaResponse) {
...
   })
 
-  const restaurants = await remoteQuery(restaurantsQuery)
-
   return res.status(200).json({ restaurants })
 }
 ```
 
-This creates a `GET` API route at `/restaurants`. It uses remote query to retrieve a restaurant, its products, and the product variant’s prices for a specified currency.
+This creates a `GET` API route at `/restaurants`. It uses the Query to retrieve a restaurant, its products, and the product variant’s prices for a specified currency.
```
Comment (olivermrbl) [inline www/apps/resources/app/recipes/marketplace/examples/restaurant-delivery/page.mdx]:
```suggestion
This creates a `GET` API route at `/restaurants`. It uses Query to retrieve a restaurant, its products, and the product variant’s prices for a specified currency.
```

File: www/apps/resources/app/recipes/marketplace/examples/restaurant-delivery/page.mdx [hunk 5]
```diff
@@ -682,7 +680,7 @@ This returns the list of restaurants in the response.
 
 ### Further Reads
 
-- [What is Remote Query and how to use it](!docs!/advanced-development/modules/remote-query)
+- [What is the Query and how to use it](!docs!/advanced-development/modules/query)
```
Comment (olivermrbl) [inline www/apps/resources/app/recipes/marketplace/examples/restaurant-delivery/page.mdx]:
```suggestion
- [What is Query and how to use it](!docs!/advanced-development/modules/query)
```

Comment (olivermrbl) [reply www/apps/resources/app/recipes/marketplace/examples/restaurant-delivery/page.mdx]: Would be great to go over it all and ensure it's Query and not the Query. Maybe we can do a search and replace

**Review by olivermrbl — APPROVED** [review]
LGTM, a few comments

File: www/apps/resources/app/recipes/marketplace/examples/restaurant-delivery/page.mdx (hunk 5 above)
Comment (shahednasser) [reply www/apps/resources/app/recipes/marketplace/examples/restaurant-delivery/page.mdx]: Yes will do, it felt weird just naming it "Query" but let's test it out 😄

**Pull request merged**